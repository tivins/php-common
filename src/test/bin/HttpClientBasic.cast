{"version":2,"width":80,"height":10,"timestamp":1674406157,"env":{"SHELL":"\/bin\/bash","TERM":"xterm-256color"}}
[0,"o","\u0007\u001b[01;32mme@hostname\u001b[00m:\u001b[01;34m\/\u001b[00m$ run.php\r\n"]
[0.40382909774780273,"o","Tivins\\Core\\Net\\Client#4 200 (542)\r\n"]
[0.40393805503845215,"o","stdClass Object\r\n(\r\n    [args] => stdClass Object\r\n        (\r\n        )\r\n\r\n    [data] => \r\n    [files] => stdClass Object\r\n        (\r\n        )\r\n\r\n    [form] => stdClass Object\r\n        (\r\n        )\r\n\r\n    [headers] => stdClass Object\r\n        (\r\n            [Accept] => *\/*\r\n            [Accept-Encoding] => deflate, gzip, br\r\n            [Content-Length] => 0\r\n            [Content-Type] => application\/x-www-form-urlencoded\r\n            [Host] => httpbin.org\r\n            [User-Agent] => Mozilla\/5.0 (X11; Ubuntu; Linux x86_64; rv:103.0) Gecko\/20100101 Firefox\/103.0\r\n            [X-Amzn-Trace-Id] => Root=1-63cd690d-0b0e5b550cc93dea5f0d9685\r\n        )\r\n\r\n    [json] => \r\n    [method] => GET\r\n    [origin] => xx.xx.xx.xx\r\n    [url] => https:\/\/httpbin.org\/anything\r\n)\r\n"]
[0.41257309913635254,"o","\u0007\u001b[01;32mme@hostname\u001b[00m:\u001b[01;34m\/\u001b[00m$ "]