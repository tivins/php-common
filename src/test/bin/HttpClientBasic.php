<?php

use Tivins\Core\Net\{ Client, ClientException };

require 'vendor/autoload.php';

try {
    $client = (new Client('https://httpbin.org/anything'))->execute();
}
catch (ClientException $ex) {
    exit($ex->client . ' : ' . $ex->getMessage() . "\n");
}

echo $client . ' ' . $client->getCode() . ' (' . strlen($client->getContent()) . ')' . PHP_EOL;
print_r($client->getContentAsJSON());
