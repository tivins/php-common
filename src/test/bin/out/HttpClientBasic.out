Tivins\Baz\Core\Net\Client#4 200 (497)
stdClass Object
(
    [args] => stdClass Object
        (
        )

    [data] => 
    [files] => stdClass Object
        (
        )

    [form] => stdClass Object
        (
        )

    [headers] => stdClass Object
        (
            [Accept] => */*
            [Content-Length] => 0
            [Content-Type] => application/x-www-form-urlencoded
            [Host] => httpbin.org
            [User-Agent] => Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:102.0) Gecko/xx.xx.xx.xx Firefox/102.0
            [X-Amzn-Trace-Id] => Root=xx.xx.xx.xx-xx.xx.xx.xxbecxx.xx.xx.xx
        )

    [json] => 
    [method] => GET
    [origin] => xx.xx.xx.xx
    [url] => https://httpbin.org/anything
)
